Sources/SimpleCurrencyConverter/SimpleCurrencyConverter.swift

```diff
@@ -1,6 +1,138 @@
-public struct SimpleCurrencyConverter {
-    public private(set) var text = "Hello, World!"
+import Foundation
 
-    public init() {
+public enum Currency: String {
+    case JPY = "JPY"
+    case USD = "USD"
+    case GBP = "GBP"
+    case EUR = "EUR"
+}
+
+public class SimpleCurrencyConverter: CurrencyConverterProtocol {
+
+    private var exchangeRatesAPIKey: String = ""
+    private var networkManager: ExchangeRatesNetworkManager?
+    private static var sharedInstance: CurrencyConverterProtocol?
+    
+    /// Public setup singleton instance method for a class that allows for the setting of an exchange rates API key and a URL session.
+    ///
+    /// If an exchange rates API key is provided, it is set to the class's exchangeRatesAPIKey property.
+    /// If no exchange rates API key is provided, the class attempts to retrieve an API key from a plist file using the ExchangeRatesAPIKey keyword.
+    ///
+    /// If a URL session is provided, it is used to initialize the class's network manager.
+    /// If no URL session is provided, the class's network manager is initialized with the shared URL session.
+
+    public static func shared() -> CurrencyConverterProtocol {
+        if sharedInstance == nil {
+            fatalError("❗ Shared instance is used without framework initialization ❗")
```

Comment:
Usually when we build a framework we try to avoid any form of aborting the (host) app in production code. So in most cases we use `assertionFailure()` and then `return` so SDK users have full control and awareness of SDK behavior.
I'm thinking if it isn't better to replace `fatalError` with `assertionFailure` and return `nil` (with changing return type to `CurrencyConverterProtocol?`). What do you think @malayli @ts-dzmitry-kurski ?

Reply:
Seems reasonable to me.